common/tools/dev-tool/src/commands/samples/checkNodeVers.ts

```diff
@@ -0,0 +1,323 @@
+import { S_IRWXO } from "constants";
+import { resolveProject } from "../../util/resolveProject";
+
+const log = createPrinter("check-node-vers-samples");
+
+function spawnCMDIgnoreError(
+  cmd: string,
+  args: string[],
+  errorMessage?: string,
+  callback?: () => void
+) {
+  const spawnedProcess = pr.spawn(cmd, args);
+  spawnedProcess.on("error", function() {
+    log.info(errorMessage);
+    callback?.();
+  });
+  spawnedProcess.on("exit", function() {
+    callback?.();
+  });
+}
+
+function deleteDockerContainerCallback(deleteContainerNames?: string[], callback?: () => void) {
+  if (deleteContainerNames) {
+    log.info(`cleanup: deleting ${deleteContainerNames.join(", ")} docker containers`);
+    spawnCMDIgnoreError(
+      "docker",
+      ["rm", ...deleteContainerNames, "-f"],
+      `Attempted to delete ${deleteContainerNames.join(
+        ", "
+      )} docker containers but encountered an error doing so`,
+      callback
+    );
+  } else {
+    callback?.();
+  }
+}
+
+function deleteDockerImageCallback(dockerImageNames?: string[], callback?: () => void) {
+  if (dockerImageNames) {
+    log.info(`cleanup: deleting ${dockerImageNames.join(", ")} docker images`);
+    spawnCMDIgnoreError(
+      "docker",
+      ["rmi", ...dockerImageNames, "-f"],
+      `Attempted to delete the ${dockerImageNames.join(
+        ", "
+      )} docker images but encountered an error doing so`,
+      callback
+    );
+  } else {
+    callback?.();
+  }
+}
+
+function deleteDockerContextCallback(dockerContextDirectory?: string, callback?: () => void) {
+  if (dockerContextDirectory) {
+    log.info(`cleanup: deleting the ${dockerContextDirectory} docker context directory`);
+    spawnCMDIgnoreError("rm", ["-rf", dockerContextDirectory], undefined, callback);
+  } else {
+    callback?.();
+  }
+}
+
+function cleanup(
+  dockerContextDirectory?: string,
+  dockerContainerNames?: string[],
+  dockerImageNames?: string[],
+  callback?: () => void
+) {
+  deleteDockerContextCallback(dockerContextDirectory, () =>
+    deleteDockerContainerCallback(dockerContainerNames, () =>
+      deleteDockerImageCallback(dockerImageNames, callback)
+    )
+  );
+}
+
+function buildRunSamplesScript(
+  containerWorkspacePath: string,
+  packageName: string,
+  envFileName: string,
+  logFilePath?: string
+) {
+  function compileCMD(cmd: string, printToScreen?: boolean) {
+    return printToScreen ? cmd : `${cmd} >> ${logFilePath} 2>&1`;
+  }
+  const printToScreen = logFilePath === undefined;
+  const packagePath = `${containerWorkspacePath}/${packageName}`;
+  const envFilePath = `${containerWorkspacePath}/${envFileName}`;
+  const javascriptSamplesPath = `${containerWorkspacePath}/samples/javascript`;
+  const typescriptCompiledSamplesPath = `${containerWorkspacePath}/samples/typescript/dist`;
+  const scriptContent = `#!/bin/sh
+
+function install_dependencies_helper() {
+\tlocal samples_path=\$1;
```

Comment:
Since you're using a multiline string literal, can you include the whitespace formatting like tabs in the string literal instead of using escape characters? It'll help this inlined document be readable.

Reply:
Done.